src/business/usecases/CreateMovieUseCase.ts

```diff
@@ -0,0 +1,43 @@
+import { Movie } from "../entities/Movie"
+import { MovieGateway } from "../gateways/MovieGateway"
+import { MissingInformationError } from "../entities/errors/MissingInformationError"
+
+export class CreateMovieUseCase {
+    private movieGateway: MovieGateway
+    constructor(
+        movieGateway: MovieGateway
+    ) {
+        this.movieGateway = movieGateway
+    }
+    async execute(input: CreateMovieUseCaseInput) {
+        if (!input.title ||
+            !input.date ||
+            !input.length ||
+            !input.synopsis ||
+            !input.link ||
+            !input.picture
+        ) {
+            throw new MissingInformationError("Faltam informações para a criação do filme!")
+        }
+
+        const movie = new Movie(
+            input.title,
+            input.date,
+            input.length,
+            input.synopsis,
+            input.link,
+            input.picture
+        );
+
+        await this.movieGateway.createMovie(movie)
```

Comment:
sei que eu fiz assim em aula, mas poderia ter passo o `new Movie` direto aqui. Assim:
`await this.movieGateway.createMovie(new Movie(...))`